src/main/java/org/javarosa/core/model/instance/ExternalDataInstanceSource.java

```diff
@@ -22,24 +22,26 @@
     TreeElement root;
     private String sourceUri;
     private String instanceId;
+    private Multimap<String, String> requestData;
```

Comment:
having this stored separately feels like a better design choice regardless of the request method since it allows more flexibility for the 'client' to decide how to serialise it.